src/Files.App.Storage/Storables/WindowsStorage/WindowsStorable.cs

```diff
@@ -1,66 +1,58 @@
 ﻿// Copyright (c) Files Community
 // Licensed under the MIT License.
 
+using System.Runtime.CompilerServices;
 using Windows.Win32;
 using Windows.Win32.Foundation;
 using Windows.Win32.System.SystemServices;
 using Windows.Win32.UI.Shell;
 
 namespace Files.App.Storage
 {
-	public abstract class WindowsStorable : IWindowsStorable, IStorableChild, IEquatable<IWindowsStorable>
+	public unsafe abstract class WindowsStorable : IWindowsStorable
 	{
-		public ComPtr<IShellItem> ThisPtr { get; protected set; }
+		public IShellItem* ThisPtr
+		{
+			[MethodImpl(MethodImplOptions.AggressiveInlining)]
+			get;
+
+			[MethodImpl(MethodImplOptions.AggressiveInlining)]
+			protected set;
+		}
```

Comment:
@dongle-the-gadget Could you review for real quick? I unwrapped the ptr-s where they are passed to constructors of WindowsFile or WindowsFolder, which means they shouldnt be disposed outside the block where defined.

Reply:
Do you have some code where the storables are used? In general you shouldn't call Release on the pointers you pass to the constructor as it looks like those objects take ownership.

Reply:
I've ensured all of the instantiation is done witout releasing.

```C#
// Instead of:

ComPtr<IShellItem> pShellItem = ...;
var folder = new WindowsFolder(pShellItem.Get());

// Did this. because no reason to use ComPtr.

IShellItem* pShellItem = ...;
var folder = new WindowsFolder(pShellItem);
```



https://github.com/files-community/Files/blob/acac1a3d35c3df31203ca02baf64d0ef9bdf5b80/src/Files.App.Storage/Storables/WindowsStorage/WindowsStorable.cs#L36-L55

https://github.com/files-community/Files/blob/acac1a3d35c3df31203ca02baf64d0ef9bdf5b80/src/Files.App/ViewModels/UserControls/Widgets/QuickAccessWidgetViewModel.cs#L235-L253

Reply:
Is everything then disposed?

Reply:
Is everything then disposed?